chirp/drivers/baofeng_uv17.py

```diff
@@ -0,0 +1,423 @@
+                    filtered += char
+                else:
+                    filtered += " "
+            return filtered
+
+        rs = RadioSetting("settings.boottext1", "Power-On Message 1",
+                          RadioSettingValueString(
+                              0, 10, _filter(self._memobj.settings.boottext1)))
+        basic.append(rs)
+
+        rs = RadioSetting("settings.boottext2", "Power-On Message 2",
+                          RadioSettingValueString(
+                              0, 10, _filter(self._memobj.settings.boottext2)))
+        basic.append(rs)
+
+        if _mem.settings.powersave > 0x04:
+            val = 0x00
+        else:
+            val = _mem.settings.powersave
+        rs = RadioSetting("settings.powersave", "Battery Saver",
+                          RadioSettingValueList(
+                              LIST_SAVE, LIST_SAVE[val]))
+        basic.append(rs)
+
+        rs = RadioSetting("settings.scanmode", "Scan Mode",
+                          RadioSettingValueList(
+                              LIST_SCANMODE,
+                              LIST_SCANMODE[_mem.settings.scanmode]))
+        basic.append(rs)
+
+        rs = RadioSetting("settings.dtmfst", "DTMF Sidetone",
+                          RadioSettingValueList(LIST_DTMFST, LIST_DTMFST[
+                              _mem.settings.dtmfst]))
+        basic.append(rs)
+
+        rs = RadioSetting("settings.fmenable", "Enable FM radio",
+                          RadioSettingValueBoolean(_mem.settings.fmenable))
+        basic.append(rs)
+
+        self.get_settings_common_dtmf(dtmfe)
+
+        return top
+
+    def decode_tone(self, val):
+        pol = "N"
+        mode = ""
+        if val in [0, 0xFFFF]:
+            xval = 0
+        elif (val & 0x8000) > 0:
+            mode = "DTCS"
+            xval = (val & 0x0f) + (val >> 4 & 0xf)\
+                * 10 + (val >> 8 & 0xf) * 100
+            if (val & 0xC000) == 0xC000:
+                pol = "R"
+        else:
+            mode = "Tone"
+            xval = int((val & 0x0f) + (val >> 4 & 0xf) * 10 +
+                       (val >> 8 & 0xf) * 100 + (val >> 12 & 0xf)
+                       * 1000) / 10.0
+
+        return mode, xval, pol
+
+    def get_memory(self, number):
+        offset = 0
+        # Skip 16 bytes at memory block boundary
+        # This is because these 16 bytes contain memory page numbers
+        if number >= 252:
+            offset += 1
+        if number >= 507:
+            offset += 1
+        if number >= 762:
+            offset += 1
+
+        _mem = self._memobj.memory[number + offset]
+        _nam = self._memobj.names[number]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        self.get_memory_common(_mem, _nam.name, mem)
+
+        return mem
+
+    def encode_tone(self, memtone, mode, tone, pol):
+        if mode == "Tone":
+            xtone = str(int(tone * 10)).rjust(4, '0')
```

Comment:
The more conventional way to do this (which is a numeric operation instead of a string one, and which avoids the pre-conversion) is:
```
xtone = '%04i' % (tone * 10)
```

Reply:
Thnx. :-) Fixed.